addons/mail/controllers/discuss/thread.py

```diff
@@ -11,3 +14,13 @@ def _filter_message_post_partners(self, thread, partners):
             # sudo: discuss.channel.member - filtering partners that are members is acceptable
             return request.env["discuss.channel.member"].sudo().search(domain).partner_id
         return super()._filter_message_post_partners(thread, partners)
+
+    @http.route("/mail/message/post", methods=["POST"], type="jsonrpc", auth="public")
+    @add_guest_to_context
+    def mail_message_post(self, thread_model, thread_id, post_data, context=None, **kwargs):
+        thread = request.env[thread_model]._get_thread_with_access(
+            thread_id, mode=request.env[thread_model]._mail_post_access, **kwargs
+        )
```

Comment:
It's not the best to duplicate the thread access logic, it would be better if there was a more specific method to override.